pythonforandroid/recipes/android/src/android/display_cutout.py

```diff
@@ -70,3 +71,32 @@ def get_size_of_bar(bar_target=None):
         bar_target = status or navigation and defaults to status
     """
     return get_width_of_bar(), get_height_of_bar(bar_target)
+
+
+def get_heights_of_both_bars():
+    " Return heights of both bars "
+    return get_height_of_bar('status'), get_height_of_bar('navigation')
+
+
+def get_cutout_mode():
+    " Return mode for cutout supported applications "
+    mode = 'never'
+
+    try:
+        LayoutParams = autoclass('android.view.WindowManager$LayoutParams')
+        window = mActivity.getWindow()
+        layout_params = window.getAttributes()
+        cutout_mode = layout_params.layoutInDisplayCutoutMode
+
+        if cutout_mode == LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_ALWAYS:
+            mode = 'always'
+        elif cutout_mode == LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT:
+            mode = 'default'
+        elif cutout_mode == LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES:
+            mode = 'shortEdges'
```

Comment:
Have you thought about a more declarative approach? See (pseudo-code) below to see what I mean:
```python
        cutout_modes = {
            LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_ALWAYS: 'always',
            LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT: 'default',
            LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES: 'shortEdges',
        }

        mode = cutout_modes.get(cutout_mode, default_mode)
```